systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java

```diff
@@ -1035,6 +1039,31 @@ void testNodePort() throws Exception {
             .done();
 
         kafkaClient.sendAndRecvMessages(NAMESPACE);
+
+        // Check that Kafka status has correct addresses in NodePort external listener part
+        for (ListenerStatus listenerStatus : KafkaResource.getKafkaStatus(CLUSTER_NAME, NAMESPACE).getListeners()) {
+            if (listenerStatus.getType().equals("external")) {
```

Comment:
You check if it has external listener but not the type. That is a bit weird. Either you expect the resource above and you do not need to check for any of them or you should check for both of them, or?

Reply:
It's used only in NodePort test so I expect that operator will create node port correctly.

Reply:
I guess you are right. I misread the code.